### PR TITLE
chore(queue): use dataclasses for EmbarkedPull

### PR DESCRIPTION
**Pull request by sileht**
Change-Id: I8103968f4e185e8e15d5d9b65bd90a2092cf7198

### CODE REVIEW

File: mergify_engine/queue/merge_train.py [hunk 1]
```diff
@@ -233,35 +277,26 @@ def deserialize(
...
                 EmbarkedPull(
+                    train,
                     data["user_pull_request_number"],  # type: ignore
                     data["config"],  # type: ignore[typeddict-item]
                     data["queued_at"],  # type: ignore[typeddict-item]
                 )
             ]
             still_queued_embarked_pulls = initial_embarked_pulls.copy()
 
-        # backward compat allow_checks_interruption ->
```
Comment (jd) [inline mergify_engine/queue/merge_train.py:253]: is this related?

Comment (jd) [reply mergify_engine/queue/merge_train.py:253]: @sileht ?

Comment (jd) [reply mergify_engine/queue/merge_train.py:253]: I still see it?

Comment (jd) [reply mergify_engine/queue/merge_train.py:253]: @sileht ?

Comment (sileht) [reply mergify_engine/queue/merge_train.py:253]: https://github.com/Mergifyio/mergify-engine/pull/4236/files#diff-8a28074908845762e6f3c7ec1d353fe5226b27bf71a33c88b3b1437ce7ff757cR142-R151

**Review by jd — APPROVED** [review]

**Comment by sileht** [comment]
@mergifyio refresh
